From 74c1812bad1d318b7eaf8c81ca300d81f35ec4d1 Mon Sep 17 00:00:00 2001
From: Hugo van Kemenade <hugovk@users.noreply.github.com>
Date: Tue, 10 May 2022 18:28:51 +0300
Subject: [PATCH 1/2] Fix ConfigParser compatibilty

The SafeConfigParser class has been renamed to ConfigParser
in Python 3.2. This alias will be removed in Python 3.12.
Use ConfigParser directly instead.

https://github.com/python/cpython/blob/75280944e5ca957eec7f814b9d0608fc84fc5811/Lib/configparser.py#L1242-L1252
---
 offlineimap/CustomConfig.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/offlineimap/CustomConfig.py b/offlineimap/CustomConfig.py
index 59da078a..758b936a 100644
--- a/offlineimap/CustomConfig.py
+++ b/offlineimap/CustomConfig.py
@@ -17,11 +17,11 @@
 import os
 import re
 from sys import exc_info
-from configparser import SafeConfigParser, Error
+from configparser import ConfigParser, Error
 from offlineimap.localeval import LocalEval
 
 
-class CustomConfigParser(SafeConfigParser):
+class CustomConfigParser(ConfigParser):
     def __init__(self):
         SafeConfigParser.__init__(self)
         self.localeval = None

From 1455770bf0322557ac1db0edb2db409b4532a47e Mon Sep 17 00:00:00 2001
From: Chris Coleman <colemanchris@hotmail.com>
Date: Sun, 14 May 2023 04:06:31 -0400
Subject: [PATCH 2/2] Update CustomConfig.py

Call the right base class, `ConfigParser` (not `SafeConfigParser`)
---
 offlineimap/CustomConfig.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/offlineimap/CustomConfig.py b/offlineimap/CustomConfig.py
index 758b936a..236f6147 100644
--- a/offlineimap/CustomConfig.py
+++ b/offlineimap/CustomConfig.py
@@ -23,7 +23,7 @@
 
 class CustomConfigParser(ConfigParser):
     def __init__(self):
-        SafeConfigParser.__init__(self)
+        ConfigParser.__init__(self)
         self.localeval = None
 
     def getdefault(self, section, option, default, *args, **kwargs):
